h2d/RenderContext.hx

```diff
@@ -305,6 +310,50 @@ class RenderContext extends h3d.impl.RenderContext {
 		baseShader.viewportB.set(viewB * flipY, viewD * flipY, viewY * flipY);
 	}
 
+	/**
+		Adds a shader the global shader list.
+
+		Global shaders are present in all draw operations along with base2d shader.
+	**/
+	public function addGlobalShader( shader : hxsl.Shader ) {
+		var list = new hxsl.ShaderList(shader, baseShaderListTail.next);
+		baseShaderListTail.next = list;
+		baseShaderListTail = list;
+		needInitShaders = true;
+	}
+
+	/**
+		Removes the shader from the global shader list.
+
+		@returns `true` if shader was removed, `false` otherwise.
+	**/
+	public function removeGlobalShader( shader : hxsl.Shader ) {
+		if ( shader == baseShader ) throw "Cannot remove base shader!";
+		var prev : hxsl.ShaderList = null;
+		var hd = baseShaderList;
+		while ( hd != baseShaderListTail.next ) {
+			if ( hd.s == shader ) {
+				prev?.next = hd.next;

```

Comment:
getting an exception on this line when compiling 
![image](https://github.com/user-attachments/assets/ffc19396-e02f-4b9c-aa69-8845978f1b9e)

Reply:
```suggestion
				if (prev != null) {
					prev.next = hd.next;
				}

```

Reply:
Not sure if that's newer Haxe or older Haxe not allowing such assignments with null syntax. Changed to `if`.